aptos-move/aptos-vm/src/move_vm_ext/session/mod.rs

```diff
@@ -76,8 +81,20 @@ impl<'r, 'l> SessionExt<'r, 'l> {
                 // We allow serialization of native values here because we want to
                 // temporarily store native values (via encoding to ensure deterministic
                 // gas charging) in block storage.
-                serialize_and_allow_delayed_values(&value, &layout)
-                    .map(|bytes| (bytes.into(), Some(Arc::new(layout))))
+                match serialize_and_allow_delayed_values_with_statement(&value, &layout) {
+                    Some((bytes, statement)) => {
+                        if statement.delayed_fields_count <= MAX_DELAYED_FIELDS_PER_RESOURCE {
+                            Some((bytes.into(), Some(Arc::new(layout))))
+                        } else {
+                            return Err(PartialVMError::new(StatusCode::TOO_MANY_DELAYED_FIELDS)
+                                .with_message(
+                                    "Delayed fields are not allowed in resource serialization."
```

Comment:
Can we actually say that there are too many delayed fields? The message is confusing right now

Reply:
Duh